### PR TITLE
[Agent] fix windows upgrade failed #20102

### PR DESCRIPTION
**Pull request by TomatoMr**
### This PR is for:

- Agent

#### Steps to reproduce the bug
- 
#### Changes to fix the bug
- 
#### Affected branches
- main
- v6.1
#### Checklist
- [ ] Added unit test to verify the fix.

### CODE REVIEW

File: agent/src/dispatcher/local_mode_dispatcher.rs [hunk 1]
```diff
@@ -295,7 +296,14 @@ impl LocalModeDispatcher {
     }
 
     pub(super) fn listener(&self) -> LocalModeDispatcherListener {
-        LocalModeDispatcherListener::new(self.base.listener(), self.extractor.clone())
+        #[cfg(target_os = "linux")]
+        {
```
Comment (rvql) [inline agent/src/dispatcher/local_mode_dispatcher.rs:300]: 不用加括号

File: agent/src/utils/environment.rs [hunk 2]
```diff
@@ -47,6 +47,27 @@ use public::utils::net::{
     get_mac_by_ip, get_route_src_ip_and_mac, link_by_name, link_list, LinkFlags, MacAddr,
 };
 
+#[cfg(target_os = "windows")]
```
Comment (rvql) [inline agent/src/utils/environment.rs:50]: 合起来

File: agent/src/dispatcher/local_mode_dispatcher.rs (hunk 1 above)
Comment (TomatoMr) [reply agent/src/dispatcher/local_mode_dispatcher.rs:300]: 不加括号，在 windows 编译时会失败

File: agent/src/utils/environment.rs (hunk 2 above)
Comment (TomatoMr) [reply agent/src/utils/environment.rs]: 已修改

File: agent/src/dispatcher/local_mode_dispatcher.rs (hunk 1 above)
Comment (rvql) [reply agent/src/dispatcher/local_mode_dispatcher.rs:300]: 就一条语句为啥会失败？

Comment (TomatoMr) [reply agent/src/dispatcher/local_mode_dispatcher.rs:300]: image
```
pub(super) fn listener(&self) -> LocalModeDispatcherListener {
        #[cfg(target_os = "linux")]
        LocalModeDispatcherListener::new(self.base.listener(), self.extractor.clone())  // 不加括号，windows 会报没有分号的错误，且不能加上分号，因为要返回 LocalModeDispatcherListener 
        #[cfg(target_os = "windows")]
        LocalModeDispatcherListener::new(self.base.listener())
    }
```

Comment (rvql) [reply agent/src/dispatcher/local_mode_dispatcher.rs:300]: 那写return应该可以？

**Review by rvql — APPROVED** [review]

**Pull request merged**